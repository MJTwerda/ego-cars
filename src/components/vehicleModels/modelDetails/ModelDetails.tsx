import { VehicleModeldetailsI } from "../interfaces/models.interfaces";
import { stripHtmlTags } from '../../../utils/stringsManagement';
import ImageTextBlock from './ImageTextBlock';
import ImageCarousel from './ImageCarousel';

interface ModelDetailsProps {
  vehicleData: VehicleModeldetailsI;
}

const ModelDetails = ({ vehicleData }: ModelDetailsProps) => {
  return (
    <div className="flex flex-col">
      <ImageTextBlock
        image={{ src: vehicleData.photo, alt: vehicleData.name, position: 'left', width: '60' }}
        textInfo={{
          model: vehicleData.name,
          title: { text: vehicleData.title, variant: 'h3' },
          description: vehicleData.description
        }}
      />

      {/* Se concatenan 3 veces para simular un carrusel con mayor visibilidad */}
      <ImageCarousel modelFeatures={vehicleData.model_features.concat(vehicleData.model_features).concat(vehicleData.model_features)} />

      <div className="flex flex-col mt-4">
        {vehicleData.model_highlights.map((highlight, index) => {
          const indexIsPar = index % 2 === 0
          return (
            <ImageTextBlock
              key={highlight.title}
              image={{ src: highlight.image, alt: highlight.title, position: indexIsPar ? 'right' : 'left' }}
              textInfo={{
                title: { text: highlight.title, variant: 'h6' },
                description: stripHtmlTags(highlight.content)
              }}
            />

          )
        })}
      </div>

    </div>
  );
};

export default ModelDetails;