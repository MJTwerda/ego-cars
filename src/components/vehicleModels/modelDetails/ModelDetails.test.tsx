import { act, render, RenderResult, waitFor } from '@testing-library/react';
import ModelDetails, { ModelDetailsProps } from './ModelDetails';

const mockProps: ModelDetailsProps = {
  vehicleData: {
    id: 12345,
    segment: 'Segment 1',
    year: 2022,
    price: 10000,
    thumbnail: '/thumbnail1.jpg',
    name: 'Model 1',
    title: 'Title 1',
    description: 'Description 1',
    photo: '/img1.jpg',
    model_features: [],
    model_highlights: []
  }
};

describe('ModelDetails', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render', async () => {
    let component: RenderResult;
    await act(async () => {
      component = render(<ModelDetails vehicleData={mockProps.vehicleData} />);
    });


    await waitFor(() => {
      expect(component).toMatchSnapshot();
      const images = component.getAllByRole('img');
      expect(images.length).toBe(1);
    });
  });

  it('should render with highlights info', async () => {
    const propsWithHighlights: ModelDetailsProps = {
      ...mockProps,
      vehicleData: {
        ...mockProps.vehicleData,
        model_highlights: [
          {
            title: 'Highlight 1',
            content: '<p>Content 1</p>',
            image: '/img1.jpg'
          },
          {
            title: 'Highlight 2',
            content: '<p>Content 2</p>',
            image: '/img2.jpg'
          }
        ]
      }
    }
    let component: RenderResult;
    await act(async () => {
      component = render(<ModelDetails vehicleData={propsWithHighlights.vehicleData} />);
    });


    await waitFor(() => {
      expect(component).toMatchSnapshot();
      const images = component.getAllByRole('img');
      expect(images.length).toBe(3);
    });
  });

});