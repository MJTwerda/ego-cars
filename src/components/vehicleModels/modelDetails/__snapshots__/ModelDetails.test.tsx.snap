// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelDetails should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
        >
          <img
            alt="Model 1"
            class="rounded lg:w-[60%] lg:mr-12"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="flex flex-col self-center lg:w-[40%]"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
            >
              Model 1
            </h5>
            <h3
              class="MuiTypography-root MuiTypography-h3 mb-4 css-13msox4-MuiTypography-root"
            >
              Title 1
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
            >
              Description 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
            </p>
          </div>
        </div>
        <div
          class="relative w-full mt-11 overflow-hidden"
        >
          <div
            class="flex p-4 sm:pr-20 transition-transform duration-700 ease-in-out bg-[#F7F7F7] gap-4"
            style="transform: translateX(-0%);"
          />
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute left-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="image_carousel_prev_button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="ArrowBackIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
              />
            </svg>
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute right-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="image_carousel_next_button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="ArrowForwardIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </button>
          <div
            class="absolute bottom-6 left-1/2 transform -translate-x-1/2 flex space-x-2"
          />
        </div>
        <div
          class="flex flex-col mt-4"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col"
    >
      <div
        class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
      >
        <img
          alt="Model 1"
          class="rounded lg:w-[60%] lg:mr-12"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <div
          class="flex flex-col self-center lg:w-[40%]"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
          >
            Model 1
          </h5>
          <h3
            class="MuiTypography-root MuiTypography-h3 mb-4 css-13msox4-MuiTypography-root"
          >
            Title 1
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
          >
            Description 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
          </p>
        </div>
      </div>
      <div
        class="relative w-full mt-11 overflow-hidden"
      >
        <div
          class="flex p-4 sm:pr-20 transition-transform duration-700 ease-in-out bg-[#F7F7F7] gap-4"
          style="transform: translateX(-0%);"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute left-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
          data-testid="image_carousel_prev_button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="ArrowBackIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
            />
          </svg>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute right-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
          data-testid="image_carousel_next_button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="ArrowForwardIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </button>
        <div
          class="absolute bottom-6 left-1/2 transform -translate-x-1/2 flex space-x-2"
        />
      </div>
      <div
        class="flex flex-col mt-4"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ModelDetails should render with highlights info 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
        >
          <img
            alt="Model 1"
            class="rounded lg:w-[60%] lg:mr-12"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="flex flex-col self-center lg:w-[40%]"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
            >
              Model 1
            </h5>
            <h3
              class="MuiTypography-root MuiTypography-h3 mb-4 css-13msox4-MuiTypography-root"
            >
              Title 1
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
            >
              Description 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
            </p>
          </div>
        </div>
        <div
          class="relative w-full mt-11 overflow-hidden"
        >
          <div
            class="flex p-4 sm:pr-20 transition-transform duration-700 ease-in-out bg-[#F7F7F7] gap-4"
            style="transform: translateX(-0%);"
          />
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute left-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="image_carousel_prev_button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="ArrowBackIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
              />
            </svg>
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute right-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
            data-testid="image_carousel_next_button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="ArrowForwardIosIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </button>
          <div
            class="absolute bottom-6 left-1/2 transform -translate-x-1/2 flex space-x-2"
          />
        </div>
        <div
          class="flex flex-col mt-4"
        >
          <div
            class="flex flex-col lg:flex-row lg:flex-row-reverse align-center mx-4 mt-11 sm:mx-32"
          >
            <img
              alt="Highlight 1"
              class="rounded lg:w-1/2 lg:ml-12"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="flex flex-col self-center lg:w-[40%]"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 mb-4 css-1miy0lu-MuiTypography-root"
              >
                Highlight 1
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
              >
                Content 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
              </p>
            </div>
          </div>
          <div
            class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
          >
            <img
              alt="Highlight 2"
              class="rounded lg:w-1/2 lg:mr-12"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=%2Fimg2.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg2.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg2.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="flex flex-col self-center lg:w-[40%]"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 mb-4 css-1miy0lu-MuiTypography-root"
              >
                Highlight 2
              </h6>
              <p
                class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
              >
                Content 2 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col"
    >
      <div
        class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
      >
        <img
          alt="Model 1"
          class="rounded lg:w-[60%] lg:mr-12"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <div
          class="flex flex-col self-center lg:w-[40%]"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-16xl4zq-MuiTypography-root"
          >
            Model 1
          </h5>
          <h3
            class="MuiTypography-root MuiTypography-h3 mb-4 css-13msox4-MuiTypography-root"
          >
            Title 1
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
          >
            Description 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
          </p>
        </div>
      </div>
      <div
        class="relative w-full mt-11 overflow-hidden"
      >
        <div
          class="flex p-4 sm:pr-20 transition-transform duration-700 ease-in-out bg-[#F7F7F7] gap-4"
          style="transform: translateX(-0%);"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute left-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
          data-testid="image_carousel_prev_button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="ArrowBackIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
            />
          </svg>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium absolute right-0 top-8 h-[150px] w-10 sm:w-12 bg-[--color-select] opacity-80 text-black hidden sm:flex items-center justify-center hover:bg-gray-600 transition rounded-none css-53g0n7-MuiButtonBase-root-MuiIconButton-root"
          data-testid="image_carousel_next_button"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
            data-testid="ArrowForwardIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </button>
        <div
          class="absolute bottom-6 left-1/2 transform -translate-x-1/2 flex space-x-2"
        />
      </div>
      <div
        class="flex flex-col mt-4"
      >
        <div
          class="flex flex-col lg:flex-row lg:flex-row-reverse align-center mx-4 mt-11 sm:mx-32"
        >
          <img
            alt="Highlight 1"
            class="rounded lg:w-1/2 lg:ml-12"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimg1.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg1.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg1.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="flex flex-col self-center lg:w-[40%]"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 mb-4 css-1miy0lu-MuiTypography-root"
            >
              Highlight 1
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
            >
              Content 1 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
            </p>
          </div>
        </div>
        <div
          class="flex flex-col lg:flex-row  align-center mx-4 mt-11 sm:mx-32"
        >
          <img
            alt="Highlight 2"
            class="rounded lg:w-1/2 lg:mr-12"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimg2.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg2.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg2.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="flex flex-col self-center lg:w-[40%]"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 mb-4 css-1miy0lu-MuiTypography-root"
            >
              Highlight 2
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 css-7ctihp-MuiTypography-root"
            >
              Content 2 Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
